Sources/Sentry/PrivateSentrySDKOnly.mm

```diff
@@ -121,27 +123,38 @@ + (NSDictionary *)getExtraContext
 }
 
 #if SENTRY_TARGET_PROFILING_SUPPORTED
-+ (uint64_t)startProfilingForTrace:(SentryId *)traceId;
++ (uint64_t)startProfilerForTrace:(SentryId *)traceId;
 {
     [SentryProfiler startWithTracer:traceId];
     return SentryDependencyContainer.sharedInstance.dateProvider.systemTime;
 }
 
-+ (nullable NSDictionary<NSString *, id> *)collectProfileForTrace:(SentryId *)traceId
-                                                            since:(uint64_t)startSystemTime;
++ (nullable NSDictionary<NSString *, id> *)collectProfileBetween:(uint64_t)startSystemTime
+                                                             and:(uint64_t)endSystemTime
+                                                        forTrace:(SentryId *)traceId;
 {
-    NSMutableDictionary<NSString *, id> *payload = [SentryProfiler
-        collectProfileBetween:startSystemTime
-                          and:SentryDependencyContainer.sharedInstance.dateProvider.systemTime
-                     forTrace:traceId
-                        onHub:[SentrySDK currentHub]];
+    NSMutableDictionary<NSString *, id> *payload =
+        [SentryProfiler collectProfileBetween:startSystemTime
+                                          and:endSystemTime
+                                     forTrace:traceId
+                                        onHub:[SentrySDK currentHub]];
 
     if (payload != nil) {
         payload[@"platform"] = SentryPlatformName;
+        payload[@"transaction"] = @{
+            @"active_thread_id" :
+                [NSNumber numberWithLongLong:sentry::profiling::ThreadHandle::current()->tid()]
```

Comment:
This is going to get the thread ID at the time that the profile is being processed, why are we overwriting the one previously set at the time the trace was started with this one? I imagine it's possible to have a different active thread for each of the 3 times: 1) when the profile is started, 2) ended and 3) being processed.

Reply:
> the one previously set at the time the trace was started

We are not, there's no thread ID stored when the profiler is started. There is one stored when a **transaction** is started, but that doesn't apply here, where we don't start objective-c transactions. And I'm aware this may be different in those scenarios, and am not completely certain it's going to be correct the way it works in Flutter but this was the simplest approach and if it works, than no need to complicate things. If it doesn't, I'll follow up. 

It's not great that I cannot test this in Flutter without the cocoa SDK being released (or without messing up with the flutter plugin build setup) - so it's a bit of a guesswork and needs a couple of iterations 🤷